STOP_SENDING opens streams (#2013)
quicwg/base-drafts: draft-ietf-quic-transport.md

data is being discarded on receipt at application request.  This signals a peer
to abruptly terminate transmission on a stream.

> Receipt of a STOP_SENDING frame is invalid for a locally-initiated stream which

Reviewer: ```suggestion
Receipt of a STOP_SENDING frame is invalid for a locally-initiated stream that
```

> does not exist or is in the "Ready" state (see {{stream-send-states}}).

Reviewer 2: "does not exist" sounds a bit weird. In my implementation, when a stream is fully closed, I delete it from my map of streams at some point. At that moment, it does not exist any more, but receipt of a (delayed) STOP_SENDING is still a valid thing.

Maybe we can say "has not yet been created".

Reviewer: ```suggestion
has not yet been created or is in the "Ready" state (see {{stream-send-states}}).
```

Receiving a STOP_SENDING frame for a locally-initiated send stream that is
"Ready" or non-existent MUST be treated as a connection error of type
PROTOCOL_VIOLATION.  An endpoint that receives a STOP_SENDING frame for a
receive-only stream MUST terminate the connection with error PROTOCOL_VIOLATION.

> 

Reviewer 3: I don't think this handles STOP_SENDING sent on an implicitly opened stream.

Reviewer 4: I think senders are required to open streams in order so this should be fine

Reviewer 4: I think senders are required to open streams in order so this should be fine
